src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,161 @@
+    function deposit(address asset, uint256 amount, address onBehalfOf, uint16 referralCode) external;
+    function withdraw(address asset, uint256 amount, address to) external;
+    function getReserveNormalizedIncome(address asset) external view returns (uint256);
+    function getReserveData(address asset) external view returns (
+        uint256,    // Configuration
+        uint128,    // the liquidity index. Expressed in ray
+        uint128,    // variable borrow index. Expressed in ray
+        uint128,    // the current supply rate. Expressed in ray
+        uint128,    // the current variable borrow rate. Expressed in ray
+        uint128,    // the current stable borrow rate. Expressed in ray
+        uint40,     // last updated timestamp
+        address,    // address of the adai interest bearing token
+        address,    // address of the stable debt token
+        address,    // address of the variable debt token
+        address,    // address of the interest rate strategy
+        uint8       // the id of the reserve
+    );
+}
+
+interface RewardsClaimerLike {
+    function claimRewards(address[] calldata assets, uint256 amount, address to) external returns (uint256);
+}
+
+contract D3MAaveDaiPool is D3MPoolBase {
+
+    uint256 constant RAY  = 10 ** 27;
+
+    address                  public immutable pool;
+    RewardsClaimerLike       public immutable rewardsClaimer;
+    ShareTokenLike           public immutable stableDebt;
+    ShareTokenLike           public immutable variableDebt;
+    address                  public immutable interestStrategy;
+    address                  public immutable adai; // Token representing a share of the asset pool
+
+    address public king;  // Who gets the rewards
+
+    event Collect(address indexed king, address[] assets, uint256 amt);
+    event File(bytes32 indexed what, address data);
+
+    constructor(address hub_, address dai_, address pool_, address _rewardsClaimer) public D3MPoolBase(hub_, dai_) {
+        pool = pool_;
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, address interestStrategy_,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPool/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPool/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPool/invalid-variableDebt");
+        require(interestStrategy_ != address(0), "D3MAaveDaiPool/invalid-interestStrategy");
+
+        adai = adai_;
+        stableDebt = ShareTokenLike(stableDebt_);
+        variableDebt = ShareTokenLike(variableDebt_);
+        interestStrategy = interestStrategy_;
+        rewardsClaimer = RewardsClaimerLike(_rewardsClaimer);
+
+        ShareTokenLike(adai_).approve(pool_, type(uint256).max);
+        TokenLike(dai_).approve(pool_, type(uint256).max);
+
+        wards[msg.sender] = 1;
+        emit Rely(msg.sender);
```

Comment:
This is being set in both the base and this contract. Should probably remove it from here.